source/common/http/utility.cc

```diff
@@ -100,6 +100,21 @@ std::string Utility::parseCookieValue(const HeaderMap& headers, const std::strin
   return state.ret_;
 }
 
+void Utility::removeCookieValues(const HeaderMap& headers, const std::string& key) {
+  headers.iterate([](const HeaderEntry& header) -> void {
+    if (header.key() == Http::Headers::get().Cookie.get().c_str()) {
+      for (const std::string& s : StringUtil::split(std::string{header.value().c_str()}, ';')) {
+        size_t first_non_space = s.find_first_not_of(" ");
+        size_t equals_index = s.find('=');
+        std::string k = s.substr(first_non_space, equals_index - first_non_space);
+        if (k == state->key_) {
```

Comment:
Secondly, does this function remove the entire Cookie header?
I am no UI expert, but isn't it the case that there is usually a single Cookie header come with multiple cookies separated by semicolon ; ?
Either way, the logic here seems to remove the entire Cookie header, while what you probably want is to remove a specific cookie right? E.g., removing cookie "foo" from the cookies. 
```bash
curl -b 'foo=bar;a=b;c' http://httpbin.org/get
{
  "args": {}, 
  "headers": {
    "Accept": "*/*", 
    "Cookie": "foo=bar;a=b;c", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.51.0"
  }, 
  "origin": "74.108.147.123", 
  "url": "http://httpbin.org/get"
}
```